nhentai/doujinshi.py

```diff
@@ -74,26 +74,28 @@ def show(self):
 
     def check_if_need_download(self, options):
         base_path = os.path.join(self.downloader.path, self.filename)
+        file_ext = None
 
-        # doujinshi directory is not exist, we need to download definitely
-        if not (os.path.exists(base_path) and os.path.isdir(base_path)):
-            return True
-
-        # regenerate, we need to re-download from nhentai
+        # regenerate, re-download
         if options.regenerate:
             return True
 
+        # detect file extensions
```

Comment:
noted that file ext will be overwrite if both `is_pdf` and `is_cbz` are specified